assignments/homework/Raymond Ingram/Homework 2.py

```diff
@@ -0,0 +1,313 @@
+                  ('M31' , '2012-01-02', 17.1, 17.4),
+                  ('M101', '2012-01-02', 15.1, 13.5),
+                  ('M82' , '2012-02-14', 16.2, 14.5),
+                  ('M31' , '2012-02-14', 16.9, 17.3),
+                  ('M82' , '2012-02-14', 15.2, 15.5),
+                  ('M101', '2012-02-14', 15.0, 13.6),
+                  ('M82' , '2012-03-26', 15.7, 16.5),
+                  ('M101', '2012-03-26', 15.1, 13.5),
+                  ('M101', '2012-03-26', 14.8, 14.3)],
+            names=['name', 'obs_date', 'mag_b', 'mag_v'])
+
+
+# In[19]:
+
+
+obs2 = obs[2::2]
+obs2
+
+
+# In[20]:
+
+
+mask1 = obs['name']=='M31'
+obs3 = obs[mask1]
+obs3
+
+
+# # <section class="challenge panel panel-success">
+# <div class="panel-heading">
+# <h2><span class="fa fa-pencil"></span> Challenge</h2>
+# </div>
+# 
+# 
+# <div class="panel-body">
+# 
+# <p>Using the <code>t6</code> table above:</p>
+# <ol>
+# <li>
+# <p>Make a plot that shows <code>j_m</code>-<code>h_m</code> on the x-axis, and <code>h_m</code>-<code>k_m</code> on the y-axis</p>
+# </li>
+# <li>
+# <p>Make a new table that contains the subset of rows where the <code>j_snr</code>, <code>h_snr</code>, and <code>k_snr</code> columns, which give the signal-to-noise-ratio in the J, H, and K band, are greater than 10, and try and show these points in red in the plot you just made.</p>
+# </li>
+# <li>
+# <p>Make a new table (based on the full table) that contains only the RA, Dec, and the <code>j_m</code>, <code>h_m</code> and <code>k_m</code> columns, then try and write out this catalog into a format that you can read into another software package. For example, try and write out the catalog into CSV format, then read it into a spreadsheet software package (e.g. Excel, Google Docs, Numbers, OpenOffice). You may run into an issue at this point - if so, take a look at https://github.com/astropy/astropy/issues/7357 to see how to fix it.</p>
+# </li>
+# </ol>
+# 
+# </div>
+# 
+# </section>
+# 
+
+# In[21]:
+
+
+from astropy.time import Time
+from astropy.coordinates import SkyCoord
+
+
+# In[22]:
+
+
+#t6 = Table()
+
+
+# In[23]:
+
+
+#t6['time'] = Time([50000, 51000, 52000], format='mjd')
+
+
+# In[24]:
+
+
+#t6['coord'] = SkyCoord([1, 2, 3] * u.deg, [4, 5, 6] * u.deg)
+
+
+# In[25]:
+
+
+#t6['flux'] = [1, 5, 4] * u.mJy
+
+
+# In[26]:
+
+
+t6 = Table.read('data/2mass.tbl', format='ascii.ipac')
+
+
+# In[27]:
+
+
+subset1 = t6['j_m']-t6['h_m']
+subset2 = t6['h_m']-t6['k_m']
+plt.scatter(subset1,subset2)
+
+
+# In[28]:
+
+
+greaterz = 10
+masker = (t6['j_snr'] > greaterz) & (t6['h_snr'] > greaterz) & (t6['k_snr'] > greaterz)
+sigToNoise = t6[masker]
+plt.scatter(subset1,subset2)
+plt.scatter(sigToNoise['j_m']-sigToNoise['h_m'],sigToNoise['h_m']-sigToNoise['k_m'], c='red')
+
+
+# In[29]:
+
+
+newTable = Table()
+newTable['ra'], newTable['dec'], newTable['j_m'], newTable['h_m'], newTable['k_m'] = t6['ra'], t6['dec'], t6['j_m'], t6['h_m'], t6['k_m']
```

Comment:
The solution works, but this can be written simpler: 

```suggestion
newTable = t6['ra', 'dec', 'j_m', 'h_m', 'k_m']
```